virtio-devices/src/watchdog.rs

```diff
@@ -96,7 +98,9 @@ impl WatchdogEpollHandler {
                 self.last_ping_time.lock().unwrap().replace(Instant::now());
             }
 
-            used_desc_heads[used_count] = (desc_chain.head_index(), len);
+            *used_desc_heads
+                .get_mut(used_count)
+                .ok_or(Error::UnexpectedAvailDescriptor)? = (desc_chain.head_index(), len);
```

Comment:
The approach is correct but let's take this as an opportunity to make the watchdog implementation follow the other virtio devices implementations. And with the new design, there will be no out-of-bound access.

Reply:
I'm gonna send a PR in a sec!